Could you check the failing test:
```python
=========================== short test summary info ============================
FAILED tests/generation/test_logits_process.py::LogitsProcessorTest::test_no_bad_words_dist_processor - AssertionError: Lists differ: [[True, True, False, True, True], [True, True, True, False, True]] != [[True, True, False, True, False], [True, True, True, False, False]]

First differing element 0:
[True, True, False, True, True]
[True, True, False, True, False]

- [[True, True, False, True, True], [True, True, True, False, True]]
?                            ^^^                              ^^^

+ [[True, True, False, True, False], [True, True, True, False, False]]
?                            ^^^^                              ^^^^
============= 1 failed, 43 passed, 4 warnings in 107.08s (0:01:47) =============
```

--- a/src/transformers/generation/logits_process.py
+++ b/src/transformers/generation/logits_process.py
@@ -1309,12 +1309,10 @@ class NoBadWordsLogitsProcessor(SequenceBiasLogitsProcessor):
                     eos_token_id = [eos_token_id]
                 eos_token_id = torch.tensor(eos_token_id)
 
-            eos_set = set(eos_token_id)
-            bad_words_ids = [
-                bad_token_seq
-                for bad_token_seq in bad_words_ids
-                if len(bad_token_seq) != 1 or bad_token_seq[0] not in eos_set
-            ]
+            eos_token_id_list = eos_token_id.tolist() # convert to python list before
+            bad_words_ids = list(
+                filter(lambda bad_token_seq: all(bad_token_seq != [i] for i in eos_token_id_list), bad_words_ids)
+            )
         # Forbidding a sequence is equivalent to setting its bias to -inf
         sequence_bias = {tuple(sequence): float("-inf") for sequence in bad_words_ids}
         super().__init__(sequence_bias=sequence_bias)

--- a/src/transformers/generation/logits_process.py
+++ b/src/transformers/generation/logits_process.py
@@ -1309,7 +1309,7 @@ class NoBadWordsLogitsProcessor(SequenceBiasLogitsProcessor):
                     eos_token_id = [eos_token_id]
                 eos_token_id = torch.tensor(eos_token_id)
 
-            eos_token_id_list = eos_token_id.tolist() # convert to python list before
+            eos_token_id_list = eos_token_id.tolist()  # convert to python list before
             bad_words_ids = list(
                 filter(lambda bad_token_seq: all(bad_token_seq != [i] for i in eos_token_id_list), bad_words_ids)
             )

